src/imageproxy.rs

```diff
@@ -394,14 +413,14 @@ impl ImageProxy {
                 rustix::net::RecvFlags::CMSG_CLOEXEC,
             )?
             .bytes;
-            let fdret = cmsg_buffer
+            let mut fdret = cmsg_buffer
                 .drain()
                 .filter_map(|m| match m {
                     rustix::net::RecvAncillaryMessage::ScmRights(f) => Some(f),
                     _ => None,
                 })
                 .flatten()
-                .next();
+                .fuse();
```

Comment:
I'd love to know why that's necessary here...

Reply:
It handles what you say below
>  It also means that you don't need to handle the impossible case of the iterator having returned None the first time and Some the second.